api/client.go

```diff
@@ -35,7 +35,8 @@ type Client struct {
 	AccessKey string
 	HTTP      http.Client
 	// Headers are HTTP headers that will be added to every request made by the Client.
-	Headers http.Header
+	Headers        http.Header
+	OnUnauthorized func()
```

Comment:
minor and not blocking: some godoc on this field would be great. Something about when it gets called so that someone using the `Client` knows how to use it.